napari/_qt/widgets/qt_highlight_preview.py

```diff
@@ -405,52 +426,81 @@ def __init__(
 
         self._refresh()
 
-    def _update_value(self, value):
-        """Update highlight value.
+    def _update_thickness_value(self, thickness_value):
+        """Update highlight thickness value.
 
         Parameters
         ----------
-        value : int
-            Highlight value.
+        thickness_value : int
+            Highlight thickness value.
         """
-        if value == '':
+        if thickness_value == '':
             return
-        value = int(value)
-        value = max(min(value, self._max_value), self._min_value)
-        if value == self._value:
+        thickness_value = int(thickness_value)
+        thickness_value = max(
+            min(thickness_value, self._max_value), self._min_value
+        )
+        if thickness_value == self._thickness_value:
             return
-        self._value = value
+        self._thickness_value = thickness_value
+        self._value['highlight_thickness'] = self._thickness_value
+        self.valueChanged.emit(self._value)
+        self._refresh()
+
+    def _update_color_value(self, color_value):
+        """Update highlight color value.
+
+        Parameters
+        ----------
+        color_value : List[float]
+            Highlight color value as a list of floats.
+        """
+        if isinstance(color_value, np.ndarray):
+            color_value = color_value.tolist()
+        if color_value == self._color_value:
+            return
+        self._color_value = color_value
+        self._value['highlight_color'] = self._color_value
         self.valueChanged.emit(self._value)
         self._refresh()
 
     def _refresh(self):
         """Set every widget value to the new set value."""
         self.blockSignals(True)
-        self._lineedit.setText(str(self._value))
-        self._slider.setValue(self._value)
-        self._triangle.setValue(self._value)
-        self._preview.setValue(self._value)
+        color = QColor(
+            *(
+                np.ceil(
+                    np.array(self._color_value) * [255, 255, 255, 255]
+                ).astype(int)[:-1]
+            )
+        )
```

Comment:
can you add a comment to explain this? I have no clue what it does at a glance xD

Reply:
Qt store RGB as sequence of 3 uint8 where vispy uses 3 floats.

Reply:
Sure! And just in case, here I'm changing the color representation from a four float list (where each list item goes from 0.0 to 1.0) to a list that has the RGB values (from 0 to 255). Also, rethinking this and besides the comment, probably it could be better to write this as:

```python
        color = QColor(
            *(
                np.ceil(
                    np.array(self._color_value)[:-1] * [255, 255, 255]
                ).astype(int)
            )
        )
```

?

Reply:
got it... I find this still a bit noisy to read and we don't really need numpy performance, so my suggestion is:

```py
        color = QColor(*[int(np.ceil(v * 255)) for v in self.color_value[:3]])
```

But feel free to ignore :)